src/Compilers/VisualBasic/Portable/Parser/Parser.vb

```diff
@@ -6078,18 +6080,30 @@ checkNullable:
         End Function
 
         Friend Function CheckFeatureAvailability(feature As Feature) As Boolean
-            Return CheckFeatureAvailability(_scanner.Options.LanguageVersion, feature)
+            Return CheckFeatureAvailability(_scanner.Options, feature)
         End Function
 
-        Friend Shared Function CheckFeatureAvailability(languageVersion As LanguageVersion, feature As Feature) As Boolean
-            Dim required = feature.GetLanguageVersion()
-            Return CInt(required) <= CInt(languageVersion)
+        Friend Shared Function CheckFeatures(feature As Feature, opts As VisualBasicParseOptions) As Boolean
+            If opts.Features Is Nothing OrElse opts.Features.Count = 0 Then Return False
+            Dim flag = feature.GetFeatureFlag
+            If flag Is Nothing Then Return False
+            Dim value As String = Nothing
+            Return opts.Features.TryGetValue(feature.GetFeatureFlag, value)
         End Function
 
-        Friend Shared Sub CheckFeatureAvailability(diagnostics As DiagnosticBag, location As Location, languageVersion As LanguageVersion, feature As Feature)
-            If Not CheckFeatureAvailability(languageVersion, feature) Then
-                Dim featureName = ErrorFactory.ErrorInfo(feature.GetResourceId())
-                diagnostics.Add(ERRID.ERR_LanguageVersion, location, languageVersion.GetErrorName(), featureName)
+        Friend Shared Function CheckFeatureAvailability(opts As VisualBasicParseOptions, feature As Feature) As Boolean
+            Dim required As LanguageVersion
+            Dim ok = feature.TryGetLanguageVersion(required)
+            Return ok AndAlso (CInt(required) <= CInt(opts.LanguageVersion)) OrElse CheckFeatures(feature, opts)
```

Comment:
Needs additional check `required <> LanguageVersion.None`, to force it to check features.